src/modals/EditModal.tsx

```diff
@@ -0,0 +1,129 @@
+import { useContext, useRef } from "react";
+import {
+  IonButton,
+  IonModal,
+  IonDatetime,
+  IonContent,
+  IonItem,
+} from "@ionic/react";
+import { useTranslation } from "react-i18next";
+import { format } from "date-fns";
+import "./EditModal.css";
+
+import { CyclesContext } from "../state/Context";
+import { useAverageLengthOfCycle } from "../state/CycleInformationHooks";
+import { getNewCyclesHistory } from "../state/CalculationLogics";
+
+interface PropsEditModal {
+  isOpen: boolean;
+  setIsOpen: (newIsOpen: boolean) => void;
+}
+
+const EditModal = (props: PropsEditModal) => {
+  const datetimeRef = useRef<null | HTMLIonDatetimeElement>(null);
+  const { t } = useTranslation();
+
+  const { cycles, updateCycles } = useContext(CyclesContext);
+  const averLengthOfCycle = useAverageLengthOfCycle();
+
+  const nowDate = new Date();
+  nowDate.setHours(0, 0, 0, 0);
+
+  const isActiveDates = (dateString: string) => {
+    if (cycles.length === 0) {
+      return true;
+    }
+    const date = new Date(dateString);
+    date.setHours(0, 0, 0, 0);
+
+    const lastCycleFinish: Date = new Date(cycles[0].startDate);
+    lastCycleFinish.setDate(lastCycleFinish.getDate() + averLengthOfCycle);
+    lastCycleFinish.setHours(0, 0, 0, 0);
+
+    return (
+      date.getTime() < lastCycleFinish.getTime() ||
+      date.getTime() <= nowDate.getTime()
+    );
+  };
+
+  function periodDays() {
+    const value: string[] = [];
+
+    for (const cycle of cycles) {
+      const startOfCycle = new Date(cycle.startDate);
+      startOfCycle.setHours(0, 0, 0, 0);
+
+      for (let i = 0; i < cycle.periodLength; i++) {
+        const newDate = new Date(startOfCycle);
```

Comment:
`newDate` is always the same on every loop step, you can move this cojst in the outer loop

Reply:
No, because in each iteration ```newDate``` is changed and written to the array.
See lines below.

Reply:
Missed this, agree